library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/mixin/KeyedResourceReloaderMixin.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.mixin;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+
+import org.spongepowered.asm.mixin.Mixin;
+
+import net.minecraft.loot.LootManager;
+import net.minecraft.recipe.RecipeManager;
+import net.minecraft.server.ServerAdvancementLoader;
+import net.minecraft.server.function.FunctionLoader;
+import net.minecraft.tag.TagManagerLoader;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.resource.loader.api.reloader.IdentifiableResourceReloader;
+import org.quiltmc.qsl.resource.loader.api.reloader.ResourceReloaderKeys;
+
+@Mixin({
+		RecipeManager.class, ServerAdvancementLoader.class, FunctionLoader.class,
+		LootManager.class, TagManagerLoader.class
+})
+public abstract class KeyedResourceReloaderMixin implements IdentifiableResourceReloader {
+	private Identifier quilt$id;
```

Comment:
Throw some `@Unique` annotations on these fields and ditch the quilt prefix

Reply:
Wished the `@Unique` annotation would make it clear which mod adds the fields instead of throwing random characters.